gcloud-java-compute/src/main/java/com/google/gcloud/compute/AddressInfo.java

```diff
@@ -0,0 +1,576 @@
+  private static final DateTimeFormatter TIMESTAMP_FORMATTER = ISODateTimeFormat.dateTime();
+
+  private final String address;
+  private final Long creationTimestamp;
+  private final String description;
+  private final String id;
+  private final AddressId addressId;
+  private final Status status;
+  private final Usage usage;
+
+  /**
+   * The status of the address.
+   */
+  public enum Status {
+
+    /**
+     * The address is reserved for the project and is available for use.
+     */
+    RESERVED,
+
+    /**
+     * The address is currently being used and thus not available.
+     */
+    IN_USE
+  }
+
+  /**
+   * Base class for a Google Compute Engine address's usage information. Implementations of this
+   * class represent different possible usages of a Compute Engine address. {@link InstanceUsage}
+   * contains information for region addresses assigned to a Google Compute Engine instance.
+   * {@link RegionForwardingUsage} contains information for region addresses assigned to one or more
+   * region forwarding rule. {@link GlobalForwardingUsage} contains information for global addresses
+   * assigned to one or more global forwarding rule.
+   */
+  public abstract static class Usage implements Serializable {
+
+    private static final long serialVersionUID = -5028609518171408695L;
+
+    Usage() {}
+
+    /**
+     * Returns the identities of resources currently using this address.
+     */
+    public abstract List<ResourceId> users();
+
+    final boolean baseEquals(Usage usage) {
+      return Objects.equals(toPb(), usage.toPb());
+    }
+
+    Address toPb() {
+      return new Address().setUsers(Lists.transform(users(), new Function<ResourceId, String>() {
+        @Override
+        public String apply(ResourceId resourceId) {
+          return resourceId.selfLink();
+        }
+      }));
+    }
+
+    @SuppressWarnings("unchecked")
+    static <T extends Usage> T fromPb(Address addressPb) {
+      String url = addressPb.getUsers().get(0);
+      if (InstanceId.matchesUrl(url)) {
+        return (T) InstanceUsage.fromPb(addressPb);
+      } else if (RegionForwardingRuleId.matchesUrl(url)) {
+        return (T) RegionForwardingUsage.fromPb(addressPb);
+      } else {
+        return (T) GlobalForwardingUsage.fromPb(addressPb);
+      }
+    }
+  }
+
+  /**
+   * Usage information for a Google Compute Engine region address assigned to a virtual machine
+   * instance.
+   */
+  public static final class InstanceUsage extends Usage {
+
+    private static final long serialVersionUID = -5028609518171408695L;
+
+    private final InstanceId instance;
+
+    InstanceUsage(InstanceId instance) {
+      this.instance = checkNotNull(instance);
+    }
+
+    /**
+     * Returns the identity of the instance using the address.
+     */
+    public InstanceId instance() {
+      return instance;
+    }
+
+    @Override
+    public List<ResourceId> users() {
+      return ImmutableList.<ResourceId>of(instance);
```

Comment:
Fine as is but did you consider returning `List<InstanceId>` (and making Usage.users returns `List<? extends ResourceId>`)?